app/code/Magento/Catalog/Api/Data/ProductInterface.php

```diff
@@ -105,6 +105,13 @@ public function setAttributeSetId($attributeSetId);
     public function getPrice();
 
     /**
+     * Product final price
+     *
+     * @return float|null
+     */
+    public function getFinalPrice();
```

Comment:
Such change is not be allowed per our backward compatibility policy: there might be implementations of this interface which will be broken.

Instead, you can implement Extension Attribute and populate it with the final price when it's loaded. Extension attribute is backward compatible change and will be allowed.

See this as a reference: app/code/Magento/Bundle/etc/extension_attributes.xml
Documentation: http://devdocs.magento.com/guides/v2.0/extension-dev-guide/attributes.html